.github/workflows/gh-pages.yml

```
@@ -0,0 +1,68 @@
+name: gh-pages
+
+on:
+    push:
+        branches:
+            - master
+            - documentation_*
```

```suggestion
    push:
        branches:
            - master
        paths:
            - "path_to_doc_src/**"
    pull_request:
        branches:
            - master
        paths:
            - "docs/**"
```
If I understand correctly you want to deploy master and a preview for PRs that touch documentation.
Instead of prescribing a branch name, my suggestion, will run on all PRs which touch a file in the documentation.

The only caveat would be if you link to code and only change the code, so maybe the paths restriction could be removed for `push` to master.

---

That's a nice solution. The master branch is now run unconditionally (links to "latest" (not yet implemented) should also reflect code changes) and PR's are published only if they touch `docs`.